scipy/special/add_newdocs.py

```diff
@@ -6374,9 +6462,39 @@ def add_newdoc(name, doc):
 
 add_newdoc("log1p",
     """
-    log1p(x)
+    log1p(x, out=None)
+
+    Calculates log(1 + x) for use when `x` is near zero.
+
+    Parameters
+    ----------
+    x : array_like
+        Real or complex valued input.
+    out : ndarray, optional
+        Optional output array for the function results.
+
+    Returns
+    -------
+    scalar or ndarray
+        Values of ``log(1 + x)``.
+
+    See Also
+    --------
+    expm1, cosm1
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using ``log(1 + x)`` directly for ``x``
+    near 0. Note that in the below example ``1 + 1e-17 == 1`` to
```

Comment:
I wonder if this is the most compelling example? It is true that the relative error is 1 for `x=1e-17`, but that might be discounted by the reader as `x` is smaller than `np.finfo(float).eps`.  For `x=1e-8` say, the relative error in using `np.log(1+x)` is `1e-10`, which is many orders of magnitude bigger than one might expect. Just a thought.